File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        arr2 = [None] * len(arr)
        maxElement = -1
        for i in range(len(arr)-1, -1, -1):
            arr2[i] = maxElement
            maxElement = max(arr[i], maxElement)

        return arr2

File: Data Structures & Algorithms/sort-colors/submission-1.java
class Solution {
    void swap(int[] arr, int i, int j) {
      int temp = arr[i];
      arr[i] = arr[j];
      arr[j] = temp;
   }

   void sortColors(int[] arr) {
      int l = 0, i = 0, r = arr.length - 1;

      while (i < r) {
         if (arr[i] == 0) {
            swap(arr, i, l);
            l++;
         } else if (arr[i] == 2) {
            swap(arr, i, r);
            r--;
            i--;
         }
         i++;
      }
   }
}